gtda/metaestimators/collection_transformer.py

```diff
@@ -0,0 +1,140 @@
+"""CollectionTransformer meta-estimator."""
+# License: GNU AGPLv3
+
+from functools import reduce
+from operator import and_
+
+import numpy as np
+from joblib import Parallel, delayed
+from sklearn.base import BaseEstimator, TransformerMixin
+from sklearn.base import clone
+from sklearn.utils.metaestimators import if_delegate_has_method
+
+from gtda.utils import check_collection
+
+
+class CollectionTransformer(BaseEstimator, TransformerMixin):
+    """Meta-transformer for applying a fit-transformer to each input in a
+    collection.
+
+    If `transformer` possesses a ``fit_transform`` method,
+    ``CollectionTransformer(transformer)`` also possesses a
+    :meth:`fit_transform` method which, on each entry in its input ``X``,
+    fit-transforms a clone of `transformer`. A collection (list or ndarray) of
+    outputs is returned.
+
+    Parameters
+    ----------
+    transformer : object
+        The fit-transformer instance from which the transformer acting on
+        collections is built. Should implement ``fit_transform``.
+
+    n_jobs : int or None, optional, default: ``None``
+        The number of jobs to use in a joblib-parallel application of
+        `transformer`'s ``fit_transform`` to each input. ``None`` means 1
+        unless in a :obj:`joblib.parallel_backend` context. ``-1`` means using
+        all processors.
+
+    parallel_backend_prefer :  ``"processes"`` | ``"threads"`` | ``None``, \
+        optional, default: ``None``
+        Soft hint for the default joblib backend to use in a joblib-parallel
+        application  of `transformer`'s ``fit_transform`` to each input. To be
+        used in conjunction with `n_jobs`. The default process-based backend is
+        "loky" and the default thread-based backend is "threading". See [1]_.
```

Comment:
So here (and in the case of mapper) I have copied the documentation for this parameter straight from joblib: https://joblib.readthedocs.io/en/latest/generated/joblib.Parallel.html

What do you suggest?

Reply:
It is strange to mention what are the default backend if they are not exposed to the user. I would remove that and simply keep the citation. And add a warning as described above.

Did you play with the `require` parameter? It might make some sense to have it too/instead.

Reply:
My interpretation of the wording in joblib is: these are the backends that result from your preferences, unless there is an explicit backend set in a global context manager. So I kinda see the point of keeping the sentence, but it's true that sending to the joblib docs is sufficient. I'll make the change.

What warning are you referring to?

Reply:
I've never played with `require` but it's a fair point that it may as well be added too (I wouldn't say instead).

Reply:
I've now added `parallel_backend_require` as you suggested.